src/Runner.Common/JobServerQueue.cs

```diff
@@ -612,6 +620,21 @@ private async Task ProcessTimelinesUpdateQueueAsync(bool runOnce = false)
                         try
                         {
                             await _jobServer.UpdateTimelineRecordsAsync(_scopeIdentifier, _hubName, _planId, update.TimelineId, update.PendingRecords, default(CancellationToken));
+                            try
+                            {
+                                if (_resultsClientInitiated)
+                                {
+                                    await _resultsServer.UpdateResultsWorkflowStepsAsync(_scopeIdentifier, _hubName, _planId, update.TimelineId, update.PendingRecords, default(CancellationToken));
+                                }
+                            }
+                            catch (Exception e)
+                            {
```

Comment:
do we want to report the error to the actions service as telemetry? like how we did for log upload?